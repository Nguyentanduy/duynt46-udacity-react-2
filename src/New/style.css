.login_container {
    display: flex;
    align-items: center;
    justify-content: center;
    height: 100vh;
    text-align: center;
}

.login {
    font-size: 30px;
    font-weight: 600;
}

.button_submit{
    margin-top: 30px;
}

input{
    padding: 10px 20px;
    width: 500px;
}

p{
    font-weight: 600;
}

button{
    font-size: 20px;
    padding: 5px 10px;
    cursor: pointer;
}