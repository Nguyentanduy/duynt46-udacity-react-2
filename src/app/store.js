import { configureStore } from '@reduxjs/toolkit';
import UserSlices from '../Slices/UserSlices';
import QuestionsSlices from '../Slices/QuestionsSlices';

const store = configureStore({
  reducer: {
    user: UserSlices,
    questions: QuestionsSlices,
  },
});

export default store