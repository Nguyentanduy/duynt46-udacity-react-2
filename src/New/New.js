import React, { useState } from 'react'
import NavBar from '../components/NavBar/NavBar'
import './style.css'
import { _saveQuestion } from '../_DATA';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { createQues } from '../Slices/UserSlices';

export default function New() {

  const [data, setData] = useState({ option1: "", option2: "" });
  const user = useSelector((state) => state.user.user);
  const navigate = useNavigate()
  const dispatch = useDispatch()

  const handleChangeInput = (e, type) => {
    setData({ ...data, [type]: e.target.value })
  }
  const handleSubmit = () => {
    if (data.option1 && data.option2) {
      _saveQuestion({ optionOneText: data.option1, optionTwoText: data.option2, author: user.id }).then((data) => {
        dispatch(createQues({ authorId: data.author, questionId: data.id }))
        navigate("/")
      })
    } else {
      alert("All input is required")
    }

  }

  return (
    <div>
      <NavBar />
      <div style={{ textAlign: 'center' }}>
        <h2>Would You Rather</h2>
        <p>Create Your Own Poll</p>
        <div>
          <div>
            <p>First Option</p>
            <input placeholder='Option One' onChange={(e) => handleChangeInput(e, "option1")} />
          </div>
          <div>
            <p>Second Option</p>
            <input placeholder='Option Two' onChange={(e) => handleChangeInput(e, "option2")} />
          </div>
          <div className='button_submit' onClick={handleSubmit}>
            <button >Submit</button>
          </div>
        </div>
      </div>
    </div>
  )
}
