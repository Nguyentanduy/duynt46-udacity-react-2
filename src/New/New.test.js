import { render } from "@testing-library/react";
import * as React from 'react';
import { MemoryRouter } from "react-router-dom";
import store from "../app/store";
import { Provider } from "react-redux";
import New from "./New";

describe('New',() => {
    it('New page should be match snapshot', () => {
        var component = render(
            <MemoryRouter>
                <Provider store={store}>
                    <New />
                </Provider>
            </MemoryRouter>
        );
        expect(component).toMatchSnapshot();
    })
})