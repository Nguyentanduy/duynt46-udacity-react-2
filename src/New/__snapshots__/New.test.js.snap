// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login New page should be match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="navbar_container"
        >
          <div
            style="display: flex;"
          >
            <div
              class="button"
            >
              Home
            </div>
            <div
              class="button"
            >
              Leaderboard
            </div>
            <div
              class="button"
            >
              New
            </div>
          </div>
          <div
            style="display: flex;"
          >
            <img
              alt="user"
              height="40"
              src="user1.png"
              width="70"
            />
            <div
              class="button"
            />
            <div
              class="button"
            >
              Logout
            </div>
          </div>
        </div>
        <hr />
        <div
          style="text-align: center;"
        >
          <h2>
            Would You Rather
          </h2>
          <p>
            Create Your Own Poll
          </p>
          <div>
            <div>
              <p>
                First Option
              </p>
              <input
                placeholder="Option One"
              />
            </div>
            <div>
              <p>
                Second Option
              </p>
              <input
                placeholder="Option Two"
              />
            </div>
            <div
              class="button_submit"
            >
              <button>
                Submit
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="navbar_container"
      >
        <div
          style="display: flex;"
        >
          <div
            class="button"
          >
            Home
          </div>
          <div
            class="button"
          >
            Leaderboard
          </div>
          <div
            class="button"
          >
            New
          </div>
        </div>
        <div
          style="display: flex;"
        >
          <img
            alt="user"
            height="40"
            src="user1.png"
            width="70"
          />
          <div
            class="button"
          />
          <div
            class="button"
          >
            Logout
          </div>
        </div>
      </div>
      <hr />
      <div
        style="text-align: center;"
      >
        <h2>
          Would You Rather
        </h2>
        <p>
          Create Your Own Poll
        </p>
        <div>
          <div>
            <p>
              First Option
            </p>
            <input
              placeholder="Option One"
            />
          </div>
          <div>
            <p>
              Second Option
            </p>
            <input
              placeholder="Option Two"
            />
          </div>
          <div
            class="button_submit"
          >
            <button>
              Submit
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`New New page should be match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="navbar_container"
        >
          <div
            style="display: flex;"
          >
            <div
              class="button"
            >
              Home
            </div>
            <div
              class="button"
            >
              Leaderboard
            </div>
            <div
              class="button"
            >
              New
            </div>
          </div>
          <div
            style="display: flex;"
          >
            <img
              alt="user"
              height="40"
              src="user1.png"
              width="70"
            />
            <div
              class="button"
            />
            <div
              class="button"
            >
              Logout
            </div>
          </div>
        </div>
        <hr />
        <div
          style="text-align: center;"
        >
          <h2>
            Would You Rather
          </h2>
          <p>
            Create Your Own Poll
          </p>
          <div>
            <div>
              <p>
                First Option
              </p>
              <input
                placeholder="Option One"
              />
            </div>
            <div>
              <p>
                Second Option
              </p>
              <input
                placeholder="Option Two"
              />
            </div>
            <div
              class="button_submit"
            >
              <button>
                Submit
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="navbar_container"
      >
        <div
          style="display: flex;"
        >
          <div
            class="button"
          >
            Home
          </div>
          <div
            class="button"
          >
            Leaderboard
          </div>
          <div
            class="button"
          >
            New
          </div>
        </div>
        <div
          style="display: flex;"
        >
          <img
            alt="user"
            height="40"
            src="user1.png"
            width="70"
          />
          <div
            class="button"
          />
          <div
            class="button"
          >
            Logout
          </div>
        </div>
      </div>
      <hr />
      <div
        style="text-align: center;"
      >
        <h2>
          Would You Rather
        </h2>
        <p>
          Create Your Own Poll
        </p>
        <div>
          <div>
            <p>
              First Option
            </p>
            <input
              placeholder="Option One"
            />
          </div>
          <div>
            <p>
              Second Option
            </p>
            <input
              placeholder="Option Two"
            />
          </div>
          <div
            class="button_submit"
          >
            <button>
              Submit
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
